invenio_communities/communities/services/components.py

```diff
@@ -69,12 +69,25 @@ class CommunityAccessComponent(ServiceComponent):
 
     def _populate_access_and_validate(self, identity, data, record, **kwargs):
         """Populate and validate the community's access field."""
-        if record is not None and "access" in data:
-            # populate the record's access field with the data already
-            # validated by marshmallow
-            record.setdefault("access", {})
-            record["access"].update(data.get("access", {}))
-            record.access.refresh_from_dict(record.get("access"))
+        is_modifying_access = record is not None and "access" in data
+
+        if not is_modifying_access:
+            return
+
+        access = data.get("access", {})
+        new_visibility = access.get("visibility")
+
+        record_has_defined_access = "access" in record
+        is_restricting = new_visibility == "restricted"
+
+        action = "manage_access" if record_has_defined_access else "create_restricted"
+
+        if record.access.visibility != new_visibility and is_restricting:
+            self.service.require_permission(identity, action, record=record)
```

Comment:
```suggestion
        record_has_defined_access = "access" in record

        action = "manage_access" if record_has_defined_access else "create_restricted"

        if record.access.visibility != new_visibility:
            self.service.require_permission(identity, action, record=record)
```

minor: I wouldn't limit only to restricting, but in general to all changes. 

Imagine e.g. an admin making a community private (because it's potentially spam/malicious), and then the community owner switching it back to public visibility.